packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -79,6 +79,15 @@ export interface EbsDeviceOptions extends EbsDeviceOptionsBase {
    * @default false
    */
   readonly encrypted?: boolean;
+
+  /**
+   * The identifier (key ID, key alias, ID ARN, or alias ARN) of the AWS Key Management Service (AWS KMS)
+   * customer master key (CMK) to use for Amazon EBS encryption.
+   * If this parameter is not specified, your AWS managed CMK for EBS is used.
+   *
+   * @default None
+   */
+  readonly kmsKeyId?: string;
```

Comment:
While I agree it might be a bit less comfortable, I think this property should be:

```suggestion
  readonly kmsKey?: IKey;
```

This provides type-safety and is more aligned with CDK patterns. This would however mean that users who want to use a key alias, or just an id, would have to resort to something like:

```typescript
const keyIdArn = Stack.of(this).formatArn({
  service: 'kms',
  resource: 'key'
  resourceName: keyId
});

const keyAliasArn = Stack.of(this).formatArn({
  service: 'kms',
  resource: 'alias'
  resourceName: aliasName
});

// then, either of these should work
const key = Key.fromKeyArn(this, 'Key', keyIdArn);
const key = Key.fromKeyArn(this, 'Key', keyAliasArn);
```

Eventually, i think we should add `fromKeyAliasName`, `fromKeyId` and `fromKeyAliasArn` to the `Key` class.

https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-kms/lib/key.ts#L305

But we can do that later in a different PR.

Reply:
Where should I do the mapping for `kmsKeyId` from `CfnInstance.BlockDeviceMappingProperty` to `key.keyId`?

Reply:
Can you elaborate? What do you mean exactly?

Reply:
I think I cannot directly replace `kmsKeyId?: string` with `kmsKey?: IKey` because the CloudFormation `Ebs` resource expects ` kmsKeyId` property. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-blockdev-template.html

As per my understanding, once we add  `kmsKey?: IKey` prop to `EbsDeviceOptions`, we will have to map `key.keyId` or `key.keyArn` to corresponding Cfn property(kmsKeyId). For example: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L1357

Please let me know if my understanding is right. 

Thanks!

Reply:
@shreyasdamle Notice that `IKey` has both a `keyId` and a `keyArn` field. 

https://github.com/aws/aws-cdk/blob/d80efae6cca60bb5c1979d28c180ef991c02eb38/packages/%40aws-cdk/aws-kms/lib/key.ts#L9-L23

Which means any place that requires a key arn or a key id can easily be replaced with an `IKey`.

Does that answer you question?

Reply:
Sorry, I'm not able to implement `IKey`, I get the following error:

`Error: Resolution error: Resolution error: Trying to resolve() a Construct at /Resources/${Token[Instance.Resource.LogicalID.2053]}/Properties/blockDeviceMappings/0/ebs/kmsKey/node/_actualNode.`

You can close this PR for now. I will do more research on this and then will reopen.

Reply:
Its ok to leave this open for now. Just let me know when you get back to it 👍